src/commands/installExtension.ts

```diff
@@ -4,9 +4,37 @@
 *--------------------------------------------------------------------------------------------*/
 
 import { IActionContext } from "@microsoft/vscode-azext-utils";
-import { commands } from "vscode";
+import { commands, extensions } from "vscode";
 
 export async function installExtension(context: IActionContext, extensionId: string): Promise<void> {
     context.telemetry.properties.extensionId = extensionId;
     await commands.executeCommand('extension.open', extensionId);
+
+    function isInstalled(): boolean {
+        return !!extensions.getExtension(extensionId);
+    }
+
+    const alreadyInstalled = isInstalled();
+    context.telemetry.properties.alreadyInstalled = String(alreadyInstalled);
+
+    if (!alreadyInstalled) {
+        return new Promise((resolve) => {
+            const disposable = extensions.onDidChange(() => {
+                const installed = isInstalled();
+                context.telemetry.properties.installedAfter30s = String(isInstalled());
+
+                if (installed) {
+                    disposable.dispose();
+                    clearTimeout(timeout);
+                    resolve();
+                }
+            });
+
+            const timeout = setTimeout(() => {
+                context.telemetry.properties.installedAfter30s = 'false';
```

Comment:
If it's been 30 seconds and they haven't installed the extension, I think it's safe to assume they didn't install it.

Reply:
Ehh, I dunno. Not everyone has fast internet. Especially for really large extensions (looking at you, storage), it could easily take longer than 30 seconds.

There's no right time limit, since internet speeds range from a 56K to multi-gigabit--so what happens when we exceed the timeout becomes important. It looks like for this, it's just telemetry?

Reply:
It's just for telemetry. And I'm open to extending the timeout. Maybe 1 or 2 minutes?

Reply:
I think the main downside to longer times is that there's a chance the event never gets sent if the user closes VSCode before download completes. I think 1 minute is reasonable; it would cover the overwhelming majority of cases, enough to give some solid data, hopefully without losing many to dropoffs.